internal/provider/platform_validations.go

```diff
@@ -26,25 +78,33 @@ func validateExternalPlatform(platform *models.Platform, cluster *common.Cluster
 		return nil
 	}
 
-	// No existing cluster or existing cluster is not external.
-	if cluster == nil || *cluster.Platform.Type != models.PlatformTypeExternal {
+	var err error
+
+	if cluster == nil || *cluster.Platform.Type != models.PlatformTypeExternal { // No existing cluster or existing cluster is not external
 		if platform.Type != nil && *platform.Type == models.PlatformTypeExternal {
 			// We require valid external settings when platform is set to external
-			return validateExternalSettingsForCreate(*platform)
+			err = validateExternalSettingsForCreate(*platform)
```

Comment:
can you write a short description for the methods:
validateExternalPlatform
validateExternalSettingsForCreate - Is it only for the create operation? what about update operation?
areExternalSettingsSet

Reply:
> we have lots of code that is specific for external platform and oci in /internal/provider/platform.go - can we move it to the external provider?

no, we cannot, it causes import cycles because the provider package is composed of other packages tightly coupled together. I think we should refactor all the packages in provider into one package (platform pakage?), it would also allow to move platform specific code in common/common.go. But, I would do that as part of another ticket/PR.

>  validateExternalSettingsForCreate - Is it only for the create operation? what about update operation?
We don't need to validate for update because the validations made at swagger level are enough.

I will add come comments :+1: